### ISSUE TITLE
Add weights to compare_means and stat_compare_means functions

### ISSUE THREAD

**Issue by 1390JSmith** [kassambara/ggpubr#255]
I have social media data that needs to be weighted by impressions. I like the ease of use of ggpubr for mean comparisons. Is there is a simple way to use weights in either function?

**Comment by xindyhu** [comment]
I agree, having the ability to allow for a weight argument will be really handy.
In practice, we often need to consider weights in calculating summary statistics and then visualize them.
I have to manually do mine using geom_signif but it's not very nice.

**Comment by kassambara** [comment]
Hi, would you please clarify this and provide an example with geom_signif?

is `weight` an argument for t.test, wilcox.test ? or is this a preprocessing step you want to apply on the data before the comparison?

thanks

**Comment by xindyhu** [comment]
@kassambara Let's say we want to answer how outcome Y differs between two groups X=0 and X = 1. X=0 group has 6 entities, and X=1 group has 18 entities. Those entities have different numbers of individuals and so we would like to consider those as weights when we compare group X = 0 and group X = 1. Right now, since compare_means in ggpubr doesn't allow me to calculate a weighted average, I have to do the process manually.
1. I calculate weighted mean and weighted standard deviation for group X=0 and X = 1.
2. I perform a weighted t.test using lm(Y~X, weights = weight), and I extract test statistics, p-values etc.
3. I create the barplot with geom_bar, geom_errorbar, geom_signif. For geom_signif, I used the manual option to specify x position, y position, annotations, etc.
4. Yesterday when I answered to this, I was having trouble with the top margin, geom_signif was too close to the top of the plot, and my annotations were all cut-off. Later I found a solution using geom_blank(). I plotted some invisible points to give me more space on top.

![image](https://user-images.githubusercontent.com/8843067/82476045-4c045600-9a82-11ea-85a5-80e76905f5a9.png)

**Comment by kassambara** [comment]
Thank you @xindyhu for this clarification! I will work on this

**Comment by kassambara** [comment]
A first-class `weights` argument is still on the list — because base `t.test()`/`wilcox.test()` don't take weights, it needs a dedicated weighted-test engine and a decision on the exact weighting semantics, so I want to get it right rather than rush it.

In the meantime you can already get a weighted comparison into a ggpubr plot today: run the weighted test with a package that supports it, then draw the bracket with `stat_pvalue_manual()`:

```r
library(ggpubr)
# your data with a weight column `w`
fit  <- lm(y ~ group, data = d, weights = w)          # or weights::wtd.t.test(), survey::svyttest()
pval <- summary(fit)$coefficients["groupB", "Pr(>|t|)"]

stat.test <- data.frame(group1 = "A", group2 = "B",
                        p = signif(pval, 3),
                        y.position = max(d$y) + 0.5)

ggbarplot(d, "group", "y", add = "mean_se", fill = "group", palette = "jco") +
  stat_pvalue_manual(stat.test, label = "p = {p}")
```

`stat_pvalue_manual()` is the supported replacement for the manual `geom_signif` step you mentioned — you just pass it a small `group1 | group2 | p | y.position` table. Keeping this issue open to track built-in weight support.

**Comment by kassambara** [comment]
Closing this. After looking into it, I've decided not to add a built-in `weight` argument to `compare_means()` / `stat_compare_means()`: the "right" weighted test depends on what the weights mean (analytic/precision weights vs frequency weights vs survey/sampling weights), and baking one interpretation into a single `weight=` option would risk giving a p-value that doesn't match a user's intended analysis.

The recommended approach is to run the weighted test that matches your weights, then annotate the plot with `stat_pvalue_manual()` — the same "compute the test, then draw it" pattern used elsewhere in ggpubr:

```r
library(ggpubr)

# analytic / precision weights (base R, no extra package)
fit  <- lm(y ~ group, data = d, weights = w)
pval <- summary(fit)$coefficients["groupB", "Pr(>|t|)"]

# survey / sampling weights -> survey::svyttest();  wtd t-test -> weights::wtd.t.test()

stat.test <- data.frame(group1 = "A", group2 = "B",
                        p = signif(pval, 3),
                        y.position = max(d$y) + 0.5)

ggboxplot(d, "group", "y") +
  stat_pvalue_manual(stat.test, label = "p = {p}")
```

This keeps the weighting semantics explicit and under your control. Happy to reopen if there's broad interest in a built-in parametric option.
